models/topics.model.js

```diff
@@ -15,19 +15,27 @@ function selectTopics(){
 }
 
 function selectArticles(id=null){
-    let sqlQueryStr = `SELECT * FROM articles`
+    let sqlQueryStr = ``
     const queryVals = []
 
     if(id){
-        sqlQueryStr += ` WHERE article_id = $1`
+        sqlQueryStr += `SELECT * FROM articles WHERE article_id = $1`
         queryVals.push(id)
+    } else {
+        sqlQueryStr += `SELECT articles.author, title, articles.article_id, topic, articles.created_at, articles.votes, article_img_url, CAST(COUNT(comments.comment_id) AS INT) AS comment_count FROM articles
+        LEFT JOIN comments ON articles.article_id = comments.article_id
+        GROUP BY articles.article_id
+        ORDER BY articles.created_at DESC;
+        `
     }
 
     return db.query(sqlQueryStr,queryVals)
     .then((result)=>{
 		if (result.rows.length === 0) {
 			return Promise.reject({ status: 404, msg: "not found" });
-		}
+		} else if (result.rows.length === 1){
```

Comment:
This else if is redundant. Your query is asking for one row, if there are no rows I agree with your promise reject. You can just return your rows[0] after the if statement. The else isn't needed because if it rejects, it'll never get there anyway. The rest of the function isn't read it's kicked right back to the catch block.

Reply:
This is where I'm wondering if it's best I separate out this function to handle selectArticles by ID and select all articles separately?  At the moment, this if..else chain is in because I've written this function to handle both scenarios (i.e. after the 'return result.rows[0]' there is a 'return result.rows' to serve back an array of all articles)